### PR TITLE
Feature/gulp structure

### PR DESCRIPTION
**Pull request by itsthatguy**

### CODE REVIEW

**Comment by rylnd** [comment]
I :heart: the refactor. Did you change any functionality? It's hard to tell from the diff.

**Comment by itsthatguy** [comment]
@rylnd yes, I added sourcemaps to css and coffeescript

File: test/test-app.js [hunk 1]
```diff
@@ -20,12 +20,65 @@ describe('firepit:app', function () {
...
+      'app/index.html',
+      'app/js/app.coffee',
+      'app/css/main.styl',
+      'app/favicon.ico'
+    ]);
+  });
+
+  it('creates data/config.yaml file', function() {
+    assert.file([
+      'data/config.yaml',
```
Comment (itsthatguy) [inline test/test-app.js]: this is *technically* invalid javascript `,` (at the end of the line)

File: test/test-app.js [hunk 2]
```diff
@@ -20,12 +20,65 @@ describe('firepit:app', function () {
...
+    ]);
+  });
+
+  it('creates scripts/ files', function() {
+    assert.file([
+      'scripts/install_dependencies.sh',
```
Comment (itsthatguy) [inline test/test-app.js]: invalid JS

File: test/test-app.js (hunk 1 above)
Comment (itsthatguy) [inline test/test-app.js]: this is *technically* invalid javascript (`,` at the end of the object)

**Pull request merged**